### PR TITLE
Don't try decrypt values on community PRs

### LINKED ISSUE #4508: Community PRs fail because of gcloud credential encryption
Currently, we have an `before_install` script that decrypts our gcloud credentials

We should check the source of the PR and then skip this step, it'll help with community PRs

### PR DESCRIPTION
**Pull request by jaxxstorm**
When a community member sends a pull request to the repo, travis can't
decrypt some environment variables which include decrypting the GCP
credentials.
In addition to this, there are a few tests that run that require the
`PULUMI_ACCESS_TOKEN` which can't be used.

This PR does 2 things:

- Adds a check to the GCP credentials decryption which checks the source
  of the PR (whether it's from within the local repo, or a fork)
- Sets an environment variable: `COMMUNITY_PRS` which is used downstream
  during some tests. If the var exists, some tests are skipped

This should enhance the experience for community pull requests
significantly

Fixes #4508

For an example of this in action, see #4507

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,8 +15,16 @@ sudo: true
 git:
   depth: false
 before_install:
-- openssl aes-256-cbc -K $encrypted_342342ee5e49_key -iv $encrypted_342342ee5e49_iv
-  -in gcp-credentials.json.enc -out gcp-credentials.json -d
+- |
+  if [ "$TRAVIS_PULL_REQUEST_SLUG" != "pulumi/pulumi" ]; 
+  then 
+    echo "Not decrypting gcp credentials";
+    export COMMUNITY_PR="true"
```
Comment (stack72) [inline .travis.yml:22]: do we need to export an environment variable? The tests won't fail if it's a community PR - the tests will skip due to not being able to read PULUMI_ACCESS_TOKEN

Comment (jaxxstorm) [reply .travis.yml:22]: Some of the tests aren't skipped even if that value isn't set, see here:

https://travis-ci.com/github/pulumi/pulumi/builds/162335371#L3290

I'm happy to set the skip to be if `PULUMI_ACCESS_TOKEN` isn't set if we think that's better?

File: pkg/cmd/pulumi/new_test.go [hunk 2]
```diff
@@ -807,6 +807,10 @@ func removeStack(t *testing.T, name string) {
 }
 
 func skipIfShort(t *testing.T) {
+	_, ok := os.LookupEnv("PULUMI_ACCESS_TOKEN")
+	if !ok {
+		t.Skipf("Skipping: PULUMI_ACCESS_TOKEN is not set")
+	}
```
Comment (lukehoban) [inline pkg/cmd/pulumi/new_test.go:813]: It's not immediately clear to me why this is right to do in `skipIfShort` - how is this check related to whether the tests are running in "short" mode?

Perhaps just the name of this function needs to change to `skipIfShortOrNoPulumiAccessToken`?

File: pkg/npm/npm_test.go [hunk 3]
```diff
@@ -33,6 +33,10 @@ func TestYarnInstall(t *testing.T) {
 }
 
 func testInstall(t *testing.T, expectedBin string) {
+	// Skip for community PRs
+	if os.Getenv("COMMUNITY_PR") == "true" {
+		t.Skip("Skipped for community PRs")
+	}
```
Comment (lukehoban) [inline pkg/npm/npm_test.go:39]: Related to the other feedback - it would be great if we could keep the codebase itself independt of the notion of "community PR" (which is really a test/CI construct).  Ideally the tests refer to semantically meaningful environment variables or parameterization - and the CI environment maps things that make sense at that layer into these in a way that makes sense.

What exactly is it that this test depends on?  Can we make the skip behaviour here more directly tied to that?

File: .travis.yml (hunk 1 above)
Comment (jaxxstorm) [reply .travis.yml]: I've switched this to use `PULUMI_ACCESS_TOKEN` instead

File: pkg/npm/npm_test.go (hunk 3 above)
Comment (jaxxstorm) [reply pkg/npm/npm_test.go:39]: It's tied to the `PULUMI_ACCESS_TOKEN`, as per the other feedback, which I think makes more sense. Does that make more sense?

File: pkg/cmd/pulumi/new_test.go (hunk 2 above)
Comment (jaxxstorm) [reply pkg/cmd/pulumi/new_test.go:813]: It's probably not, I was really just trying to take advantage of the existing function being available.

I've updated the name as suggested

File: pkg/npm/npm_test.go (hunk 3 above)
Comment (jaxxstorm) [reply pkg/npm/npm_test.go]: Actually, I'm not even sure why I added this, I've removed it.
